packages/financial-templates-lib/src/clients/FinancialContractClient.ts

```diff
@@ -315,25 +361,28 @@ class FinancialContractClient {
   // becomes: numTokens * 10^sD * trv * 10^trvD * collateralRequirement * 10^18 > amountCollateral * 10^cD.
   // To accommodate these different decimal points we can normalize each term to the 10^18 basis and then apply a
   // "correction" factor due to the additional scalling from multiplying basis numbers.
-  _isUnderCollateralized(numTokens, amountCollateral, tokenRedemptionValue) {
+  private _isUnderCollateralized(numTokens: string, amountCollateral: string, tokenRedemptionValue: string) {
     // Normalize the inputs. Now all terms are 18 decimal delimited and no extra conversion is needed.
     const normalizedNumTokens = this.normalizeSyntheticDecimals(numTokens);
     const normalizedAmountCollateral = this.normalizeCollateralDecimals(amountCollateral);
     const normalizedTokenRedemptionValue = this.normalizePriceFeedDecimals(tokenRedemptionValue);
 
+    if (!isDefined(this.collateralRequirement)) throw new Error("initialSetup hasn't been called");
```

Comment:
Makes sense. I was mainly just copying the surrounding code's style as to not change anything. But we can build official style recommendations around this if you think it makes sense!

Reply:
cool, yea i dont think we need to take any action necessarily it would be too much to refact, and this should just be a direct translation

Reply:
ye, I agree we should standardize. @daywiss has called me out on this before and I think it's a great point to make as it can make debugging harder if we do things differently, over and above a desire for just general consistency.

Case and point are how we catch errors in the serverless hub. ideally, we throw the same way everywhere so we can catch the same way (in a perfect world we can catch all types of errors but that's not always the case).